libnavigation-base/src/main/java/com/mapbox/navigation/base/session/NavigationSession.kt

```diff
@@ -0,0 +1,9 @@
+package com.mapbox.navigation.base.session
+
+interface NavigationSession {
```

Comment:
Do we need to expose this in the `base` module? If we only want to use it for internal purposes in `core, let's keep it this way.

Reply:
I thought it would be a good idea to expose this because there are situations where app developers want to listen to state changes as to whether we are actively navigating or not. Exposing for such observer to state changes help them to directly subscribe to it rather than hacking their way out to find out when are we navigating or not

Reply:
See https://github.com/mapbox/mapbox-navigation-android/pull/2462#discussion_r378821137 for my thoughts. But whether or not we expose the observers, this interface looks unnecessary to me. Simply exposing register/unregister methods from `MapboxNavigation` should be enough.